### ISSUE TITLE
Mishandling of imports

### ISSUE THREAD

**Issue by FasterSpeeding** [microsoft/pyright#1838]
**Describe the bug**
When importing a sub-module along with it's parent module in one statement (e.g. `import logging.config`) pyright is failing to detect loaded attributes on the parent module.

**To Reproduce**
Lint the example code and watch it complain about "getLogger" and "DEBUG" not being known members of the module typing then run the code and watch it not throw any attribute errors at run time. 
```
C:\Users\Lucina\PycharmProjects\hikari\test2.py
  C:\Users\Lucina\PycharmProjects\hikari\test2.py:3:18 - error: "getLogger" is not a known member of module (reportGeneralTypeIssues)
  C:\Users\Lucina\PycharmProjects\hikari\test2.py:4:25 - error: "DEBUG" is not a known member of module (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 infos
```

As a note, adding `import logging` above `import logging.config` to the following example code "fixes" the pyright errors but is also a redundant import.

**Expected behavior**
Since all the module attributes this calls are available at runtime pylint should pass without any errors when run on this code.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
import logging.config

logger = logging.getLogger()  # pylint complains about "getLogger" not being a known member
logger.setLevel(logging.DEBUG)  # pylint complains about "DEBUG" not being a known member
```

**VS Code extension or command-line**
Running pyright 1.1.138 on Windows 10 Pro (21364.1000)

**Comment by erictraut** [comment]
This is intended behavior. Your code should not rely on the side effects of a parent module's attributes being populated when explicitly importing a submodule. You should explicitly import each module that you use.
